src/platform/packages/shared/kbn-esql-validation-autocomplete/src/validation/commands/rrf/index.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { ESQLAst, ESQLCommand, ESQLMessage } from '@kbn/esql-ast';
+import { i18n } from '@kbn/i18n';
+import { ReferenceMaps } from '../../types';
+
+export function validate(
+  command: ESQLCommand<'rrf'>,
+  references: ReferenceMaps,
+  ast: ESQLAst
+): ESQLMessage[] {
+  const messages: ESQLMessage[] = [];
+
+  if (!isRrfImmediatelyAfterFork(ast)) {
+    messages.push({
+      location: command.location,
+      text: i18n.translate(
+        'kbn-esql-validation-autocomplete.esql.validation.rrfMissingScoreMetadata',
+        {
+          defaultMessage: '[RRF] Must be immediately preceded by a FORK command.',
+        }
+      ),
+      type: 'error',
+      code: 'rrfNotImmediatelyAfterFork',
+    });
+  }
+
+  if (!references.fields.get('_id')) {
+    messages.push(buildMissingMetadataMessage(command, '_id', 'Id'));
+  }
+
+  if (!references.fields.get('_index')) {
+    messages.push(buildMissingMetadataMessage(command, '_index', 'Index'));
+  }
+
+  if (!references.fields.get('_score')) {
+    messages.push(buildMissingMetadataMessage(command, '_score', 'Score'));
+  }
+
+  return messages;
+}
+
+function buildMissingMetadataMessage(
+  command: ESQLCommand<'rrf'>,
+  metadataField: string,
+  fieldName: string
+): ESQLMessage {
+  return {
+    location: command.location,
+    text: i18n.translate('kbn-esql-validation-autocomplete.esql.validation.rrfMissingMetadata', {
+      defaultMessage: `[RRF] {metadataField} metadata must be selected in the FROM command.`,
+      values: { metadataField },
+    }),
+    type: 'error',
+    code: `rrfMissing${fieldName}Metadata`,
```

Comment:
You dont need a different code per field. 
```suggestion
    code: `rrfMissingMetadataField`,
```

Reply:
✅ 37b15c765ff0ff61603f6f42328d96a54f77cbd0